algebra-core fails to compile on nightly

It seems that recent version of `algebra-core` cannot be compiled on nightly.
error looks like this
```
error[E0277]: cannot multiply-assign `&&<P as quadratic_extension::QuadExtParameters>::BaseField` to `<P as quadratic_extension::QuadExtParameters>::BaseField`
Error:    --> /home/runner/.cargo/git/checkouts/zexe-f74c42809fa6cfe1/85cac17/algebra-core/src/fields/models/quadratic_extension.rs:135:28
    |
135 |         self.c0.mul_assign(&element);
    |                            ^^^^^^^^ no implementation for `<P as quadratic_extension::QuadExtParameters>::BaseField *= &&<P as quadratic_extension::QuadExtParameters>::BaseField`
    |
    = help: the trait `std::ops::MulAssign<&&<P as quadratic_extension::QuadExtParameters>::BaseField>` is not implemented for `<P as quadratic_extension::QuadExtParameters>::BaseField`
help: consider further restricting the associated type
    |
134 |     pub fn mul_assign_by_basefield(&mut self, element: &P::BaseField) where <P as quadratic_extension::QuadExtParameters>::BaseField: std::ops::MulAssign<&&<P as quadratic_extension::QuadExtParameters>::BaseField> {
    |                                                                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
here's the [error log](https://github.com/arkworks-rs/sumcheck/runs/1228605985) if you have access to it

--- a/algebra-core/src/curves/models/short_weierstrass_jacobian.rs
+++ b/algebra-core/src/curves/models/short_weierstrass_jacobian.rs
@@ -441,7 +441,7 @@ impl<P: Parameters> ProjectiveCurve for GroupProjective<P> {
             let d = ((self.x + &b).square() - &a - &c).double();
 
             // E = 3*A
-            let e = a + a.double_in_place();
+            let e = a + &*a.double_in_place();
 
             // F = E^2
             let f = e.square();
@@ -454,7 +454,7 @@ impl<P: Parameters> ProjectiveCurve for GroupProjective<P> {
             self.x = f - &d - &d;
 
             // Y3 = E*(D-X3)-8*C
-            self.y = (d - &self.x) * &e - c.double_in_place().double_in_place().double_in_place();
+            self.y = (d - &self.x) * &e - &*c.double_in_place().double_in_place().double_in_place();
             self
         } else {
             // http://www.hyperelliptic.org/EFD/g1p/auto-shortw-jacobian-0.html#doubling-dbl-2009-l
@@ -483,7 +483,7 @@ impl<P: Parameters> ProjectiveCurve for GroupProjective<P> {
             self.x = t;
             // Y3 = M*(S-T)-8*YYYY
             let old_y = self.y;
-            self.y = m * &(s - &t) - yyyy.double_in_place().double_in_place().double_in_place();
+            self.y = m * &(s - &t) - &*yyyy.double_in_place().double_in_place().double_in_place();
             // Z3 = (Y1+Z1)^2-YY-ZZ
             self.z = (old_y + &self.z).square() - &yy - &zz;
             self
@@ -644,7 +644,7 @@ impl<'a, P: Parameters> AddAssign<&'a Self> for GroupProjective<P> {
             self.x = r.square() - &j - &(v.double());
 
             // Y3 = r*(V - X3) - 2*S1*J
-            self.y = r * &(v - &self.x) - (s1 * &j).double_in_place();
+            self.y = r * &(v - &self.x) - &(s1 * &j).double();
 
             // Z3 = ((Z1+Z2)^2 - Z1Z1 - Z2Z2)*H
             self.z = ((self.z + &other.z).square() - &z1z1 - &z2z2) * &h;

--- a/algebra-core/src/fields/models/quadratic_extension.rs
+++ b/algebra-core/src/fields/models/quadratic_extension.rs
@@ -132,8 +132,8 @@ impl<P: QuadExtParameters> QuadExtField<P> {
     }
 
     pub fn mul_assign_by_basefield(&mut self, element: &P::BaseField) {
-        self.c0.mul_assign(&element);
-        self.c1.mul_assign(&element);
+        self.c0.mul_assign(element);
+        self.c1.mul_assign(element);
     }
 }
 

--- a/crypto-primitives/src/nizk/gm17/constraints.rs
+++ b/crypto-primitives/src/nizk/gm17/constraints.rs
@@ -285,12 +285,12 @@ where
                 P::G2Var::new_variable(r1cs_core::ns!(cs, "h_beta"), || Ok(pvk.h_beta), mode)?;
             let g_alpha_pc = P::G1PreparedVar::new_variable(
                 r1cs_core::ns!(cs, "g_alpha_pc"),
-                || Ok(pvk.g_alpha.into()),
+                || Ok(E::G1Prepared::from(pvk.g_alpha)),
                 mode,
             )?;
             let h_beta_pc = P::G2PreparedVar::new_variable(
                 r1cs_core::ns!(cs, "h_beta_pc"),
-                || Ok(pvk.h_beta.into()),
+                || Ok(E::G2Prepared::from(pvk.h_beta)),
                 mode,
             )?;
             let g_gamma_pc = P::G1PreparedVar::new_variable(
@@ -345,9 +345,9 @@ where
             let h_beta_g2 =
                 P::G2Var::new_variable(r1cs_core::ns!(cs, "h_beta"), || Ok(vk.h_beta_g2), mode)?;
             let g_gamma_g1 =
-                P::G1Var::new_variable(r1cs_core::ns!(cs, "g_gamma"), || Ok(&vk.g_gamma_g1), mode)?;
+                P::G1Var::new_variable(r1cs_core::ns!(cs, "g_gamma"), || Ok(vk.g_gamma_g1), mode)?;
             let h_gamma_g2 =
-                P::G2Var::new_variable(r1cs_core::ns!(cs, "h_gamma"), || Ok(&vk.h_gamma_g2), mode)?;
+                P::G2Var::new_variable(r1cs_core::ns!(cs, "h_gamma"), || Ok(vk.h_gamma_g2), mode)?;
             let query =
                 Vec::new_variable(r1cs_core::ns!(cs, "query"), || Ok(vk.query.clone()), mode)?;
             Ok(Self {

--- a/gm17/src/r1cs_to_sap.rs
+++ b/gm17/src/r1cs_to_sap.rs
@@ -5,7 +5,7 @@ use crate::Vec;
 use core::ops::Deref;
 use r1cs_core::{ConstraintSystemRef, SynthesisError};
 
-use core::ops::{AddAssign, SubAssign};
+use core::ops::AddAssign;
 #[cfg(feature = "parallel")]
 use rayon::prelude::*;
 
@@ -161,9 +161,9 @@ impl R1CStoSAP {
         let d1_double = d1.double();
         let mut h: Vec<E::Fr> = vec![d1_double; domain_size];
         cfg_iter_mut!(h).zip(&a).for_each(|(h_i, a_i)| *h_i *= a_i);
-        h[0].sub_assign(&d2);
+        h[0] -= d2;
         let d1d1 = d1.square();
-        h[0].sub_assign(&d1d1);
+        h[0] -= &d1d1;
         h.push(d1d1);
 
         domain.coset_fft_in_place(&mut a);

--- a/r1cs-std/src/bits/uint.rs
+++ b/r1cs-std/src/bits/uint.rs
@@ -76,7 +76,7 @@ macro_rules! make_uint {
                         tmp >>= 1;
                     }
 
-                    $name {
+                    Self {
                         bits,
                         value: Some(value),
                     }
@@ -139,7 +139,7 @@ macro_rules! make_uint {
                         .cloned()
                         .collect();
 
-                    $name {
+                    Self {
                         bits: new_bits,
                         value: self
                             .value
@@ -165,7 +165,7 @@ macro_rules! make_uint {
                         .map(|(a, b)| a.xor(b))
                         .collect::<Result<_, _>>()?;
 
-                    Ok($name {
+                    Ok(Self {
                         bits,
                         value: new_value,
                     })
@@ -192,7 +192,7 @@ macro_rules! make_uint {
 
                     // Compute the maximum value of the sum so we allocate enough bits for
                     // the result
-                    let mut max_value = (operands.len() as u128) * u128::from($native::max_value());
+                    let mut max_value = (operands.len() as u128) * u128::from(core::$native::MAX);
 
                     // Keep track of the resulting value
                     let mut result_value = Some(0u128);
@@ -252,7 +252,7 @@ macro_rules! make_uint {
                         // We can just return a constant, rather than
                         // unpacking the result into allocated bits.
 
-                        return Ok($name::constant(modular_value.unwrap()));
+                        return Ok(Self::constant(modular_value.unwrap()));
                     }
                     let cs = operands.cs();
 
@@ -285,7 +285,7 @@ macro_rules! make_uint {
                     // Discard carry bits that we don't care about
                     result_bits.truncate($size);
 
-                    Ok($name {
+                    Ok(Self {
                         bits: result_bits,
                         value: modular_value,
                     })

--- a/r1cs-std/src/macros.rs
+++ b/r1cs-std/src/macros.rs
@@ -47,7 +47,7 @@ macro_rules! impl_bounded_ops {
             #[tracing::instrument(target = "r1cs", skip(self))]
             #[allow(unused_braces)]
             fn $fn(self, other: Self) -> Self::Output {
-                $impl(self, other)
+                ($impl)(self, other)
             }
         }
 
@@ -127,7 +127,7 @@ macro_rules! impl_bounded_ops {
             #[tracing::instrument(target = "r1cs", skip(self))]
             #[allow(unused_braces)]
             fn $fn(self, other: $native) -> Self::Output {
-                $constant_impl(self, other)
+                ($constant_impl)(self, other)
             }
         }
 

